### PR TITLE
[FE](mysql) support mysql protocol with precision and scale info for decimal type

### PR DESCRIPTION
**Pull request by zhangstar333**
now if use jdbc mysql driver connect to doris,
when the sql query of result has decimal type,
and user want to get the column meta info, eg: resultSetMetaData.getPrecision(i)   resultSetMetaData.getScale(i)
the precision and scale info it's wrong for the column

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.84 seconds
 stream load tsv:          441 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230331072232_clickbench_pr_123288.html

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java [hunk 1]
```diff
@@ -253,17 +255,53 @@ private int getMysqlTypeLength(PrimitiveType type) {
...
+            case DECIMAL64:
+            case DECIMAL128: {
+                // f.getDecimals() > 0 ? clampedGetLength(f) - 1 + f.getPrecisionAdjustFactor() :
+                // clampedGetLength(f) + f.getPrecisionAdjustFactor();
+                ScalarType decimalType = (ScalarType) type;
+                int precision = decimalType.decimalPrecision();
+                int scale = decimalType.decimalScale();
+                if (scale > 0) {
+                    precision += 2;
+                } else {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java:275]: why scale == 0 need +=1 ？

File: fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java [hunk 2]
```diff
@@ -253,17 +255,53 @@ private int getMysqlTypeLength(PrimitiveType type) {
...
+                    precision += 1;
+                }
+                return precision;
+            }
+            case VARCHAR:
+            case CHAR: {
+                //lengthInBytes / f.getMaxBytesPerCharacter();
+                return 3 * ((ScalarType) type).getLength();
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java:283]: the length may be -1 in some case make sure is right ?

**Comment by zhangstar333** [comment]
run p1

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run clickbench

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**